src/main/java/io/odpf/firehose/sink/mongodb/MongoSink.java

```diff
@@ -0,0 +1,105 @@
+package io.odpf.firehose.sink.mongodb;
+
+import com.mongodb.BulkWriteError;
+import com.mongodb.BulkWriteException;
+import com.mongodb.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.WriteModel;
+import io.odpf.firehose.consumer.Message;
+import io.odpf.firehose.exception.NeedToRetry;
+import io.odpf.firehose.metrics.Instrumentation;
+import io.odpf.firehose.sink.AbstractSink;
+import io.odpf.firehose.sink.mongodb.request.MongoRequestHandler;
+import org.bson.Document;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.odpf.firehose.metrics.Metrics.SINK_MESSAGES_DROP_TOTAL;
+
+/**
+ * MongoDB sink
+ */
+public class MongoSink extends AbstractSink {
+    private MongoRequestHandler mongoRequestHandler;
+    private List<WriteModel<Document>> bulkRequest;
+    private long mongoRequestTimeoutInMs;
+    private List<String> mongoRetryStatusCodeBlacklist;
+    private MongoCollection<Document> mongoCollection;
+    private MongoClient mongoClient;
```

Comment:
let's make fields final, if they are not being set again.

Reply:
fixed